lib/environment.rb

```diff
@@ -450,7 +451,7 @@ def clean_up
       @user_manager.clean_up if @user_manager
       @hosts.each {|h| h.clean_up } if @hosts
       if @service_project
-        @service_project.delete_graceful(by: nil)
+        @service_project.ensure_deleted(user: nil)
```

Comment:
THe original idea behind `delete_graceful` was not to wait for project to disappear because next scenario may nor even need service project at all. So it will have time to disappear in the background. Is this change needed to avoid problems? Or is it something you did just for the peace of mind?

Reply:
the current behaviour is that the the project stays in `phase: Terminating` for so long that the next test already starts to use it and fails to create debug pod because the project is terminating.

```
  STDERR:
  Error from server (Forbidden): pods "qeci-3419-8s8xx-control-plane-0-debug" is forbidden: unable to create new content in namespace tests-cucushift-oc44-standard-slave-gtknl-0 because it is being terminated
```

We could keep this lazy delete and then just use the `wait_for_disapppear` before the `project.create`.

Reply:
well I guess the other problem is that `delete_graceful` doesn't clear the project cache.

Reply:
retesting without `ensure_deleted`

Reply:
latest test without `ensure_deleted`

http://pastebin.test.redhat.com/847052